### PR TITLE
Towards better testing for routing, using a canned gossmap from my mainnet node.

### PR DESCRIPTION
**Pull request by rustyrussell**
This fixes the gossmap-compress tool, places a compressed gossmap snapshot from my node in the tree (only 500k!) and then does some simply optimization to askrene's min-cost-flow solver based on a very simple python call to call getroutes on it.

### CODE REVIEW

File: plugins/askrene/mcf.c [hunk 1]
```diff
@@ -541,16 +540,13 @@ static void linear_network_add_adjacenct_arc(
 		const u32 node_idx,
 		const struct arc arc)
 {
-	assert(arc.idx < tal_count(linear_network->arc_tail_node));
+	assert(arc.idx < linear_network->max_num_arcs);
 	linear_network->arc_tail_node[arc.idx] = node_idx;
 
-	assert(node_idx < tal_count(linear_network->node_adjacency_first_arc));
+	assert(node_idx < linear_network->max_num_arcs);
```
Comment (Lagrang3) [inline plugins/askrene/mcf.c:546]:
```suggestion
	assert(node_idx < linear_network->max_num_nodes);
```

File: plugins/askrene/mcf.c [hunk 2]
```diff
@@ -413,7 +412,7 @@ static struct arc node_adjacency_begin(
 		const struct linear_network * linear_network,
 		const u32 node)
 {
-	assert(node < tal_count(linear_network->node_adjacency_first_arc));
+	assert(node < linear_network->max_num_arcs);
```
Comment (Lagrang3) [inline plugins/askrene/mcf.c:415]:
```suggestion
	assert(node < linear_network->max_num_nodes);
```

File: plugins/askrene/mcf.c (hunk 1 above)
Comment (Lagrang3) [reply plugins/askrene/mcf.c:546]: If the name begins with `node_` it indicates a node property, an array as big as the number of nodes.

Comment (rustyrussell) [reply plugins/askrene/mcf.c:546]: Doh!  It's even called "node_idx"!

**Pull request merged**